src/theory/quantifiers/sygus/cegis_core_connective.h

```diff
@@ -0,0 +1,386 @@
+   * Return whether this module has the possibility to construct solutions. This
+   * is true if this module has been initialized, and the shape of the
+   * conjecture allows us to use the core connective algorithm.
+   */
+  bool isActive() const;
+
+ protected:
+  /** do cegis-implementation-specific initialization for this class */
+  bool processInitialize(Node conj,
+                         Node n,
+                         const std::vector<Node>& candidates,
+                         std::vector<Node>& lemmas) override;
+  /** do cegis-implementation-specific post-processing for construct candidate
+   *
+   * satisfiedRl is whether all refinement lemmas are satisfied under the
+   * substitution { enums -> enum_values }.
+   */
+  bool processConstructCandidates(const std::vector<Node>& enums,
+                                  const std::vector<Node>& enum_values,
+                                  const std::vector<Node>& candidates,
+                                  std::vector<Node>& candidate_values,
+                                  bool satisfiedRl,
+                                  std::vector<Node>& lems) override;
+
+  /** construct solution
+   *
+   * This function is called when candidates -> candidate_values is the current
+   * candidate solution for the synthesis conjecture.
+   *
+   * If this function returns true, then this class adds to solv the
+   * a candidate solution for candidates.
+   */
+  bool constructSolution(const std::vector<Node>& candidates,
+                         const std::vector<Node>& candidate_values,
+                         std::vector<Node>& solv);
+
+ private:
+  /** common constants */
+  Node d_true;
+  Node d_false;
+  /** The first-order datatype variable for the function-to-synthesize */
+  TNode d_candidate;
+  /**
+   * Information about the pre and post conditions of the synthesis conjecture.
+   * This maintains all information needed for producing solutions relative to
+   * one direction of the synthesis conjecture. In other words, this component
+   * may be focused on finding a C1 ... Cn such that A => C1 V ... V Cn
+   * or alteratively C1 ^ ... ^ Cn such that C1 ^ ... ^ Cn => B.
+   */
+  class Component
+  {
+   public:
+    Component() : d_numFalseCores(0), d_numRefPoints(0) {}
+    /** initialize
+     *
+     * This initializes this component with pre/post condition given by n
+     * and sygus constructor c.
+     */
+    void initialize(Node n, Node c);
+    /** get the formula n that we initialized this with */
+    Node getFormula() const { return d_this; }
+    /** Is this component active? */
+    bool isActive() const { return !d_scons.isNull(); }
+    /** Add term n to pool whose sygus analog is s */
+    void addToPool(Node n, Node s);
+    /** Add a refinement point to this component */
+    void addRefinementPt(Node id, const std::vector<Node>& pt);
+    /** Add a false case to this component */
+    void addFalseCore(Node id, const std::vector<Node>& u);
```

Comment:
what is a false core?

Reply:
I've added documentation above the d_falseCore member.